## Project 2 - Residency 2 - Statistical Learning

In [1]:
import numpy as np
import pandas as pd
%matplotlib inline
titan_df = pd.read_csv('titaninsurance.csv')
print(titan_df.head())
titan_df.describe()

   SALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)
0            1                         57                         62
1            2                        103                        122
2            3                         59                         54
3            4                         75                         82
4            5                         84                         84


SALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)
count    30.000000                  30.000000                  30.000000
mean     15.500000                  68.033333                  72.033333
std       8.803408                  20.455980                  24.062395
min       1.000000                  28.000000                  32.000000
25%       8.250000                  54.000000                  55.000000
50%      15.500000                  67.000000                  74.000000
75%      22.750000                  81.500000                  85.750000
max      30.000000                 110.000000                 122.000000

### 1. Find the mean of old scheme and new scheme column. (5 points)

In [2]:
titan_df.mean()

SALESPERSON                  15.500000
Old Scheme (in thousands)    68.033333
New Scheme (in thousands)    72.033333
dtype: float64

In [3]:
titan_df.columns

Index(['SALESPERSON', 'Old Scheme (in thousands)',
       'New Scheme (in thousands)'],
      dtype='object')

In [4]:
mean_old_scheme = titan_df[['Old Scheme (in thousands)']].mean()
print("Mean of", mean_old_scheme)

Mean of Old Scheme (in thousands)    68.033333
dtype: float64


In [5]:
mean_new_scheme = titan_df[['New Scheme (in thousands)']].mean()
print("Mean of", mean_new_scheme)

Mean of New Scheme (in thousands)    72.033333
dtype: float64


### 2. Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs? (10 points)

In [6]:
from scipy.stats import ttest_1samp, ttest_ind, mannwhitneyu, levene, shapiro

In [7]:
titan_df[['Old Scheme (in thousands)']].head()

Old Scheme (in thousands)
0                         57
1                        103
2                         59
3                         75
4                         84

In [8]:
titan_df[['New Scheme (in thousands)']].head()

New Scheme (in thousands)
0                         62
1                        122
2                         54
3                         82
4                         84

In [9]:
t_statistic, p_value = ttest_ind(titan_df[['Old Scheme (in thousands)']], titan_df[['New Scheme (in thousands)']])
print(t_statistic, p_value)

[-0.69370676] [0.49063516]


#### Null Hypothesis : There is no significant difference between data sets - Old Scheme and New Scheme

#### Alternate Hypothesis : There is a significant difference

In [10]:
# p_value < 0.05 (alpha) => alternative hypothesis:
print ("two-sample t-test p-value=", p_value)

two-sample t-test p-value= [0.49063516]


##### p_value is 0.49063516 which is greater than 0.05 (alpha), so don't reject the Null Hypothesis. i.e. there is no significant difference between old scheme and new scheme.

### 3. What conclusion does the test (p-value) lead to? (2.5 points)

p_value is 0.49063516 which is greater than 0.05, so there is no significant difference between old scheme and new scheme (they have the same mean at the 5% significance level)

### 4. Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme. If this figure is alternative hypothesis, what is:
4.1 The probability of a type 1 error? (2.5 points)

4.2 What is the p- value of the hypothesis test if we test for a difference of $5000? (10 points)

4.3 Power of the test (5 points)

#### 4.1 The probability of a type 1 error? (2.5 points)


In [19]:
# probability of type 1 error is alpha.
prob_type_1_err = 0.05
print(prob_type_1_err)


0.05


#### 4.2 What is the p- value of the hypothesis test if we test for a difference of $5000? (10 points)

In [12]:
titan_df2 = titan_df
titan_df2["New_Output (in thousands)"] = titan_df[['New Scheme (in thousands)']] + 5

In [13]:
titan_df2.head()

SALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)  \
0            1                         57                         62   
1            2                        103                        122   
2            3                         59                         54   
3            4                         75                         82   
4            5                         84                         84   

   New_Output (in thousands)  
0                         67  
1                        127  
2                         59  
3                         87  
4                         89

In [14]:
titan_df2.mean()

SALESPERSON                  15.500000
Old Scheme (in thousands)    68.033333
New Scheme (in thousands)    72.033333
New_Output (in thousands)    77.033333
dtype: float64

In [15]:
t_statistic, p_value = ttest_ind(titan_df2[['Old Scheme (in thousands)']], titan_df2[['New_Output (in thousands)']])
print(t_statistic, p_value)

[-1.56084021] [0.1240018]


In [16]:
# p_value < 0.05 (alpha) => alternative hypothesis:
print ("two-sample t-test p-value=", p_value)

two-sample t-test p-value= [0.1240018]


#### The p-value = [0.1240018]

#### 4.3 Power of the test (5 points)

In [17]:
titan_df2.describe()

SALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)  \
count    30.000000                  30.000000                  30.000000   
mean     15.500000                  68.033333                  72.033333   
std       8.803408                  20.455980                  24.062395   
min       1.000000                  28.000000                  32.000000   
25%       8.250000                  54.000000                  55.000000   
50%      15.500000                  67.000000                  74.000000   
75%      22.750000                  81.500000                  85.750000   
max      30.000000                 110.000000                 122.000000   

       New_Output (in thousands)  
count                  30.000000  
mean                   77.033333  
std                    24.062395  
min                    37.000000  
25%                    60.000000  
50%                    79.000000  
75%                    90.750000  
max                   127.000000

In [18]:
mu = 77.033333
sigma = 24.062395
t_stat = -1.56084021
p_val = 0.1240018
beta = 
print("Power of the test: ", 1 - beta)